core/src/main/java/org/infinispan/remoting/transport/jgroups/SingleResponseFuture.java

```diff
@@ -43,7 +47,7 @@ public void setTimeoutFuture(Future<?> timeoutFuture) {
    @Override
    public Void call() throws Exception {
       // The request timed out
-      complete(request.getResult());
+      complete(new Rsp<>());
```

Comment:
A Rsp with `!isReceived() && !isSuspected()` is already interpreted as a timeout, so I thought I'd avoid having another check (and adding the exception message to the Log interface ;)). Unfortunately it's harder to do the same for `RspListFuture`, so maybe I *should* throw the exception here...